runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -80,19 +85,21 @@ decl_module! {
 		}
 
 		/// Include backed candidates and bitfields.
-		#[weight = (1_000_000_000, DispatchClass::Mandatory)]
+		#[weight = (
+			MINIMAL_INCLUSION_INHERENT_WEIGHT + backed_candidates.len() as Weight * BACKED_CANDIDATE_WEIGHT,
+			DispatchClass::Mandatory,
+		)]
 		pub fn inclusion(
 			origin,
 			signed_bitfields: SignedAvailabilityBitfields,
 			backed_candidates: Vec<BackedCandidate<T::Hash>>,
 			parent_header: Header,
-		) -> DispatchResult {
+		) -> DispatchResultWithPostInfo {
 			ensure_none(origin)?;
 			ensure!(!<Included>::exists(), Error::<T>::TooManyInclusionInherents);
 
 			// Check that the submitted parent header indeed corresponds to the previous block hash.
-			let now = <frame_system::Module<T>>::block_number();
-			let parent_hash = <frame_system::Module<T>>::block_hash(now - One::one());
+			let parent_hash = <frame_system::Module<T>>::parent_hash();
```

Comment:
@pepyakin I'm pretty sure this change is functionally equivalent, but it makes testing possible, which it isn't when looking up the parent hash by block number. Do you agree?

Reply:
ah, yeah, for sure